lib/http_server/response/response.ex

```diff
@@ -24,4 +24,10 @@ defmodule HttpServer.Response.Response do
       404 => "Not Found"
     }[code]
   end
+
+  def update_headers(response) when response.body != nil do
+    %{response | headers: %{response.headers | content_length: byte_size(response.body)}}
+  end
+
+  def update_headers(response), do: response
```

Comment:
(this is a non-blocking comment)

When is it better to use guards and when is it better to pattern match?

For example, here is an alternate solution using pattern matching:
```suggestion
  def update_headers(%Response{body: nil} = response), do: response

  def update_headers(%Response{headers: headers, body: body} = response) do
    %Response{response | headers: %{headers | content_length: byte_size(body)}}
  end

  
```

Reply:
We use pattern match to check if some data (variables, maps, structs etc.) matches a specific pattern.

Guards are more powerful that pattern match. They uses pattern matching to achieve more complex checks: 
> [Guards](https://hexdocs.pm/elixir/master/patterns-and-guards.html) are a way to augment pattern matching with more complex checks. They are allowed in a predefined set of constructs where pattern matching is allowed, such as function definitions, case clauses, and others.

I think in our case it is enough to use only patterns. I'll make changes.